README.md

```diff
@@ -19,10 +19,94 @@ Create a `BanyanDBClient` with host, port and a user-specified group and then es
 ```java
 // use `default` group
 client = new BanyanDBClient("127.0.0.1", 17912, "default");
-// establish a connection
+// to send any request, a connection to the server must be estabilished
 client.connect(channel);
 ```
 
+## Schema Management
+
+### Stream
+
+Then we may define a stream with customized configurations,
+
+```java
+StreamMetadataRegistry streamRegistry = client.streamRegistry();
+// build a stream "sw" with 2 shards and ttl equals to 30 days
+Stream s = new Stream("sw", 2, Duration.ofDays(30));
+s.addTagNameAsEntity("service_id").addTagNameAsEntity("service_instance_id").addTagNameAsEntity("state");
+// TagFamily - data
+TagFamilySpec dataFamily = new TagFamilySpec("data");
+dataFamily.addTagSpec(TagFamilySpec.TagSpec.newBinaryTag("data_binary"));
+s.addTagFamilySpec(dataFamily);
+// TagFamily - searchable
+TagFamilySpec searchableFamily = new TagFamilySpec("searchable");
+searchableFamily.addTagSpec(TagFamilySpec.TagSpec.newStringTag("trace_id"))
+        .addTagSpec(TagFamilySpec.TagSpec.newIntTag("state"))
+        .addTagSpec(TagFamilySpec.TagSpec.newStringTag("service_id"));
+s.addTagFamilySpec(searchableFamily);
+// create with the stream schema
+streamRegistry.create(s);
+```
+
+with `StreamMetadataRegistry`, CRUD operations are supported for `Stream`, `IndexRuleBinding` and `IndexRule`.
+
+### IndexRuleBinding/IndexRule
+
+For better search performance, index rules are necessary for `Stream` while `IndexRuleBinding` helps 
+bind the `IndexRule` to the `Stream`,
+
+```java
+IndexRuleMetadataRegistry indexRuleRegistry = client.indexRuleRegistry();
+// create IndexRule with inverted index type and save it to series store
+IndexRule indexRule = new IndexRule("db.instance", IndexRule.IndexType.INVERTED, IndexRule.IndexLocation.SERIES);
+// tag name specifies the indexed tag
+indexRule.addTag("db.instance");
+// create the index rule "db.instance"
+this.client.create(indexRule);
+```
+
+and then create an `IndexRuleBinding` to bind it/them to the Stream,
```

Comment:
Is this binding `1:1`? If so, I think the client API doesn't have to follow in this way. It is better to provide API as simple as possible. 

`this.client.define("sw", "sw-index-rule-binding", indexRule)`. This is super easier.

Reply:
> Is this binding `1:1`? If so, I think the client API doesn't have to follow in this way. It is better to provide API as simple as possible.

The relation between Stream/Measure and IndexRuleBinding is 1:1 while IndexRuleBinding and IndexRule are 1:N

> `this.client.define("sw", "sw-index-rule-binding", indexRule)`. This is super easier.

What do you mean by `define(streamName/measureName, ruleBindingName, indexRule)`?

Reply:
My point is, whether should we provide a simple API for user about exposing index creation. The binding and rule are for implementation mostly, I think. The end users usually just add one operation. @hanahmily Could you provide some suggestions?

Reply:
> My point is, whether should we provide a simple API for user about exposing index creation. The binding and rule are for implementation mostly, I think. The end users usually just add one operation. @hanahmily Could you provide some suggestions?

Probably we could provide some shorthand APIs to simplify schema creation, i.e. we make the schema POJOs as rich domain models.

```java
Stream stream = client.define(Stream);
// and Measure measure = client.define(Measure);
IndexRuleBinding binding = stream.defineIndexRuleBinding(IndexRuleBinding);
binding.define(List indexRules);
// Or we may even skip indexRuleBinding creation
stream.defineIndexRules(List indexRules);
```

Reply:
I prefer to skip `binding` creation. But let's wait for @hanahmily for a minute.

Reply:
there would be more than one binding for a rule under a rolling upgrade. but only one binding will be active at the same time. 

If the java client doesn't intend to leverage a rolling upgrade, simply dropping the existing rule then adding a new rule instead, I'm free to use neat and shorthand APIs.

Reply:
OK, I think we should skip rolling upgrade for now. Let's keep a simple version of APIs for now.